src/software/thunderscope/robot_communication.py

```diff
@@ -229,14 +245,20 @@ def __enter__(self):
             self.multicast_channel + "%" + self.interface, PRIMITIVE_PORT, True
         )
 
+        self.running = True
+
         self.send_estop_state_thread.start()
-        self.run_world_thread.start()
         self.run_primitive_set_thread.start()
 
+        if self.run_worlds:
+            self.run_world_thread.start()
```

Comment:
Do we need `run_worlds`? It would make more sense to have the thread just start up in `setup_for_fullsystem`

Reply:
oh ok, so it's fine if all the thread start stuff isn't in the enter method? Since we're sure we're going to call setup fullsystem after we enter